plugins/sql-language/src/main/java/org/polypheny/db/sql/SqlProcessor.java

```diff
@@ -0,0 +1,427 @@
+        final StopWatch stopWatch = new StopWatch();
+        if ( log.isDebugEnabled() ) {
+            log.debug( "Planning Statement ..." );
+        }
+        stopWatch.start();
+
+        Config sqlToAlgConfig = NodeToAlgConverter.configBuilder().build();
+        final RexBuilder rexBuilder = new RexBuilder( statement.getTransaction().getTypeFactory() );
+
+        final AlgCluster cluster = AlgCluster.create( statement.getQueryProcessor().getPlanner(), rexBuilder, null, statement.getDataContext().getSnapshot() );
+        final Config config =
+                NodeToAlgConverter.configBuilder()
+                        .config( sqlToAlgConfig )
+                        .trimUnusedFields( false )
+                        .convertTableAccess( false )
+                        .build();
+        final SqlToAlgConverter sqlToAlgConverter = new SqlToAlgConverter( validator, statement.getTransaction().getSnapshot(), cluster, StandardConvertletTable.INSTANCE, config );
+        AlgRoot logicalRoot = sqlToAlgConverter.convertQuery( context.getQueryNode().orElseThrow(), false, true );
+
+        // Decorrelate
+        final AlgBuilder algBuilder = config.getAlgBuilderFactory().create( cluster, null );
+        logicalRoot = logicalRoot.withAlg( AlgDecorrelator.decorrelateQuery( logicalRoot.alg, algBuilder ) );
+
+        // Trim unused fields.
+        if ( RuntimeConfig.TRIM_UNUSED_FIELDS.getBoolean() ) {
+            logicalRoot = trimUnusedFields( logicalRoot, sqlToAlgConverter );
+        }
+
+        if ( log.isTraceEnabled() ) {
+            log.trace( "Logical query plan: [{}]", AlgOptUtil.dumpPlan( "-- Logical Plan", logicalRoot.alg, ExplainFormat.TEXT, ExplainLevel.DIGEST_ATTRIBUTES ) );
+        }
+        stopWatch.stop();
+        if ( log.isDebugEnabled() ) {
+            log.debug( "Planning Statement ... done. [{}]", stopWatch );
+        }
+
+        return logicalRoot;
+    }
+
+
+    @Override
+    public void unlock( Statement statement ) {
+        LockManager.INSTANCE.unlock( List.of( LockManager.GLOBAL_LOCK ), (TransactionImpl) statement.getTransaction() );
+    }
+
+
+    @Override
+    public void lock( Statement statement ) throws DeadlockException {
+        LockManager.INSTANCE.lock( List.of( Pair.of( LockManager.GLOBAL_LOCK, LockMode.EXCLUSIVE ) ), (TransactionImpl) statement.getTransaction() );
+    }
+
+
+    @Override
+    public String getQuery( Node parsed, QueryParameters parameters ) {
+        return ((SqlNode) parsed).toSqlString( PolyphenyDbSqlDialect.DEFAULT ).getSql();
+    }
+
+
+    @Override
+    public AlgDataType getParameterRowType( Node sqlNode ) {
+        return validator.getParameterRowType( sqlNode );
+    }
+
+
+    public List<String> splitStatements( String statements ) {
+        return SqlUtil.splitStatements( statements );
+    }
+
+
+    // Add default values for unset fields
+    private void addDefaultValues( Transaction transaction, SqlInsert insert ) {
```

Comment:
This should definitely not be part of the SQL processor.... We need to refactor and improve the whole default values logic in the next iteration.

Reply:
Agreed.